### PR TITLE
PRMDR 89: infrastructure set up prior to ticket start

### PR DESCRIPTION
**Pull request by SRAlexander**
…a files

### CODE REVIEW

File: infrastructure/lambda-search-doc-references.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ]
+}
+
+module "search-document-references-lambda" {
+  source  = "./modules/lambda"
+  name    = "SearchDocumentReferencesLambda"
+  handler = "handlers.search_patient_details_handler.lambda_handler"
+  iam_role_policies = [
+    "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+    "arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy"
```
Comment (NogaNHS) [inline infrastructure/lambda-search-doc-references.tf:29]: This lambda is going to need permissions to access the S3 bucket and dynamoDB:
    module.document_reference_dynamodb_table.dynamodb_policy,
    module.ndr-document-store.s3_object_access_policy,

File: infrastructure/lambda-search-doc-references.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  ]
+  rest_api_id       = aws_api_gateway_rest_api.ndr_doc_store_api.id
+  resource_id       = module.search-document-references-gateway.gateway_resource_id
+  http_method       = "GET"
+  api_execution_arn = aws_api_gateway_rest_api.ndr_doc_store_api.execution_arn
+  depends_on = [
+    aws_api_gateway_rest_api.ndr_doc_store_api,
+    module.search-document-references-gateway
+  ]
+}
```
Comment (NogaNHS) [inline infrastructure/lambda-search-doc-references.tf:39]: missing environment variables for s3 bucket and dynamo table

File: infrastructure/lambda-search-doc-references.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
+module "search-document-references-gateway" {
```
Comment (NogaNHS) [inline infrastructure/lambda-search-doc-references.tf:1]: the lambda and gateway needs to be referenced in 'aws_api_gateway_deployment' in triggers and depends_on

**Review by SRAlexander — COMMENTED** [review]
Checked additional configurations for Lambda, all look good from @NogaNHS

**Review by NogaNHS — APPROVED** [review]

**Pull request merged**